### PR TITLE
Pull in upstream 0.3.3-0.3.7 changes

### PR DESCRIPTION
**Pull request by mdentremont**
Key change is the re-adding of proper enum support

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/aioTV/django-rest-swagger/pull/17" target="_blank">ReviewNinja

File: rest_framework_swagger/introspectors.py [hunk 1]
```diff
@@ -508,10 +532,7 @@ def get_data_type(field):
     # elif isinstance(field, fields.SlugField):
         # return 'string', 'string', # 'slug'
     elif isinstance(field, fields.ChoiceField):
-        first_key = list(field.choices)[0]
-        if isinstance(first_key, int):
-            return 'integer', 'int64'
-        return 'string', 'string'
+        return 'choice', 'choice'
```
Comment (rygwdn) [inline rest_framework_swagger/introspectors.py:536]: `choice` is not a valid data type in swagger 2.0/openapi: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#data-types

Comment (mdentremont) [inline rest_framework_swagger/introspectors.py:536]: See L814 -- choice isn't actually used

Comment (rygwdn) [inline rest_framework_swagger/introspectors.py:536]: it was replaced by the `enum` key on params: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#parameter-object

Comment (rygwdn) [inline rest_framework_swagger/introspectors.py:536]: it's a bit odd.. WBN to have a comment here to that effect

**Comment by rygwdn** [comment]
+1

File: rest_framework_swagger/introspectors.py (hunk 1 above)
Comment (mdentremont) [inline rest_framework_swagger/introspectors.py:536]: See latest commit

File: rest_framework_swagger/introspectors.py [hunk 2]
```diff
@@ -488,6 +499,19 @@ def build_query_parameters_from_django_filters(self):
...
 
 
+def get_primitive_type(var):
+    if isinstance(var, bool):
+        return 'boolean', 'boolean'
+    elif isinstance(var, int):
+        return 'integer', 'int64'
+    elif isinstance(var, float):
+        return 'number', 'float'
+    elif isinstance(var, six.string_types):
```
Comment (dpatterson-aio) [inline rest_framework_swagger/introspectors.py:509]: is this on python3 now too? if so, probably don't need six

Comment (mdentremont) [inline rest_framework_swagger/introspectors.py:509]: This code is from upstream, I would rather leave it as close to vanilla as possible to reduce future merge pains

**Comment by mdentremont** [comment]
Re-adding Ryan's +1 --- I added a comment to address his optional concern

**Comment by dpatterson-aio** [comment]
+1

**Pull request merged**